src/python/pants/backend/helm/target_types.py

```diff
@@ -105,3 +110,86 @@ class HelmChartFieldSet(FieldSet):
     chart: HelmChartMetaSourceField
     sources: HelmChartSourcesField
     dependencies: HelmChartDependenciesField
+
+
+class AllHelmChartTargets(Targets):
+    pass
+
+
+@rule
+def all_helm_chart_targets(all_targets: AllTargets) -> AllHelmChartTargets:
+    return AllHelmChartTargets([tgt for tgt in all_targets if HelmChartFieldSet.is_applicable(tgt)])
+
+
+# -----------------------------------------------------------------------------------------------
+# `helm_unittest_test` target
+# -----------------------------------------------------------------------------------------------
+
+
+class HelmUnitTestChartField(SpecialCasedDependencies):
+    alias = "chart"
+    help = f"Dependency on a `{HelmChartTarget.alias}` target that provides with templates for the Helm unit tests."
```

Comment:
> This dependency has been implemented in a field that extends `SpecialCasedDependencies` (ideally it should be a `SingleSpecialCasedDependency` kind of field, as there can only be one). This could be a bit niche, although a field of this kind could also be useful for a future implementation of the `helm_deployment` target, as that will need a single dependency on a chart too.

So, the big question for me is: "why subclass `SpecialCasedDependencies`?" I apologize for the strange naming there, but: the idea behind that type was to allow dependencies which should be added for the purposes of change detection, but which would _not_ be matched by default in `TransitiveTargetsRequest`.

...to try to answer my own question: it seems like maybe you're using it as a workaround for the fact that we don't currently allow multiple `Dependencies` fields to be added to one target? Ideally we would change that (cc @Eric-Arellano for thoughts on how feasible that is, and/or to create a ticket), but in the mean time, it seems like the better option would be to declare a single `Dependencies` field, and then add as many sets of inference `@rules` as you need to supply the different "types" of dependencies.

Reply:
+1 o Stu's point.

There should be a number of examples in the code base of filtering a list of targets by a `FieldSet` (via the `.is_applicable` method) or possessing a specific field (via `.has_field` method on `Target`).

Reply:
so, it definetly is the case that I missunderstood the original purpose of that type. I'll consolidate this in the single `Dependencies` field as suggested.

Reply:
Yeah `SpecialCasedDependencies` is weird, it's mostly used to handle the `archive` target with the `files` and `packages` fields, along with `relocated_files`. They're all like `dependencies`, so should show up in `./pants dependencies`, but there are reasons we don't want to use normal `dependencies`:

1) For `archive`, we want two distinct fields and `dependencies` can't be more than one per target
2) For both targets, we do not want `TransitiveTargets` to apply. When you depend on a `relocated_files` target and do things like `Await Get(TransitiveTargets)`, we usually don't want to include the original files targets..we only want the relocated files.